### PR TITLE
Add support for Symfony 5

### PR DESCRIPTION
**Pull request by Big-Shark**

### CODE REVIEW

**Comment by Big-Shark** [comment]
@Jean85 Done

**Comment by kralmichal** [comment]
Hi, Symfony 5.0 came out today.  Can we merge this?

**Comment by Big-Shark** [comment]
@Jean85 Your move )

**Comment by Jean85** [comment]
Sorry I'm at SymfonyCon so I didn't had the time to check it :smile: But I saw the release live :tada:

I'm going to re-start the CI to see if everything is ok....

**Comment by rvanlaak** [comment]
We had the following error after installing it on a Symfony 5 project, so an additional forward-compatibility fix seems needed:

image

**Comment by Jean85** [comment]
We're missing an integration test, that's why we missed this :cry: 

I'll need to setup a test like that before going forward.

**Comment by Jean85** [comment]
Symfony 5 is still not getting installed :cry: There's something amiss with the CI config. I'll try to force it somehow.

**Comment by Jean85** [comment]
Conflict solved, build is failing :(

**Comment by DavidBadura** [comment]
Hey, thank you for trying to push this PR forward! <3 I'm trying to help:

1) You have requested a non-existent parameter "kernel.root_dir". Did you mean one of these: "kernel.project_dir", "kernel.cache_dir", "kernel.logs_dir"?

You need to change `kernel.root_dir` to `kernel.project_dir`. `kernel.root_dir` is deprecated since SF4.2 and removed in SF5.0. `kernel.project_dir` exists since SF3.3.

2) Could not reflect class Symfony\Component\Console\Event\ConsoleCommandEvent as it is marked final.

You do not have to mock value objects. You can just instantiate them. These include:
* Request
* Token
* *Event like ConsoleCommandEvent

Of course, you can mock the service away.


If you want, I can try to finish this PR on the weekend.

**Comment by Big-Shark** [comment]
@DavidBadura yes please, do it.

**Comment by Jean85** [comment]
Thanks @DavidBadura for the suggestions, I've implemented it. I've also removed some pieces of code that were needed for Symfony < 3.3.

**Comment by Jean85** [comment]
I should've fixed all the tests. Now onto fixing PHPStan.

**Comment by Vincz** [comment]
@Jean85 Hey! The remaining errors seems related to the PHP Unit version. The current version is 8.5 while the bundle uses the 7.5

**Comment by Jean85** [comment]
In reality I did a mistake with the class aliases. I fear that something else is still missing, let's see the CI after this fix...

**Comment by cleentfaar** [comment]
@Jean85 need help?

**Comment by Jean85** [comment]
I'm currently getting deprecation errors under PHP 7.4, probably from PHPUnit. I'll try to use the PHPUnit bridge to allow PHPUnit 8

**Comment by Jean85** [comment]
Ok two other issues:
 * I'm still getting deprecations under 7.4, I can't trace them :( any suggestion? https://travis-ci.org/getsentry/sentry-symfony/jobs/622739268#L787-L788 
 * @garak your `sed` is "updating" the PHPUnit bridge too, how can I ignore it too? https://travis-ci.org/getsentry/sentry-symfony/jobs/622739265#L643

**Comment by garak** [comment]
@Jean85 I see that [phpunit-bridge](https://packagist.org/packages/symfony/phpunit-bridge) has "normal" versions, why do you want to ignore it in sed replace?

**Comment by garak** [comment]
@Jean85 deprecation could be probably solved by allowing phpunit 8

File: composer.json [hunk 1]
```diff
@@ -36,13 +36,13 @@
         "php-http/mock-client": "^1.0",
         "phpstan/phpstan-shim": "^0.11",
         "phpstan/phpstan-phpunit": "^0.11",
-        "phpunit/phpunit": "^7.5",
-        "symfony/browser-kit": "^3.4||^4.0",
-        "symfony/expression-language": "^3.4||^4.0",
-        "symfony/framework-bundle": "^3.4||^4.0",
+        "phpunit/phpunit": "^7.5||^8.5",
```
Comment (jderusse) [inline composer.json]: remove this, and let `symfony/phpunit-bridge` pick the right version `./vendor/bin/simple-phpunit`

File: test/EventListener/RequestListenerTest.php [hunk 2]
```diff
@@ -10,23 +10,25 @@
...
 class RequestListenerTest extends BaseTestCase
 {
+    use SetUpTearDownTrait;
+
     private $currentScope;
     private $currentHub;
     private $options;
 
-    protected function setUp()
+    protected function doSetUp(): void
```
Comment (jderusse) [inline test/EventListener/RequestListenerTest.php]: not needed, because the projet require php ^7.1n you can saftly add the typehint here `setUp(): void`

**Comment by Jean85** [comment]
Because the trait is missing. But as per https://github.com/getsentry/sentry-symfony/pull/266#discussion_r355595491, I may not need it. BTW the PHPUnit bridge has no requirements, so you do not need to downgrade it.

**Comment by Jean85** [comment]
PHP 7.4 deprecations will be fixed by https://github.com/getsentry/sentry-php/pull/930 upstream.

**Comment by rvanlaak** [comment]
@Jean85 planning on releasing soon? If not I'll verify by requiring `master` locally.

**Comment by Jean85** [comment]
Normally @HazAT is in charge of releasing. Feel free to test it out!

**Pull request merged**